tasks/src/test/kotlin/TaskRunnerTest.kt

```diff
@@ -0,0 +1,197 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.tasks
+
+import com.typesafe.config.ConfigFactory
+
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.extensions.system.withEnvironment
+import io.kotest.matchers.collections.shouldContainExactly
+import io.kotest.matchers.collections.shouldContainExactlyInAnyOrder
+
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.just
+import io.mockk.mockkStatic
+import io.mockk.runs
+import io.mockk.slot
+import io.mockk.unmockkStatic
+
+import java.io.IOException
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentMap
+
+import org.eclipse.apoapsis.ortserver.dao.test.mockDatabaseModule
+import org.eclipse.apoapsis.ortserver.dao.test.unmockDatabaseModule
+import org.eclipse.apoapsis.ortserver.dao.test.verifyDatabaseModuleIncluded
+
+import org.koin.core.Koin
+import org.koin.core.context.startKoin
+import org.koin.core.context.stopKoin
+import org.koin.core.module.Module
+import org.koin.core.qualifier.named
+import org.koin.dsl.module
+import org.koin.test.KoinTest
+
+class TaskRunnerTest : KoinTest, WordSpec() {
+    init {
+        beforeTest {
+            mockDatabaseModule()
+        }
+
+        afterTest {
+            unmockDatabaseModule()
+        }
+
+        "runTasks" should {
+            "execute a single task" {
+                val taskLog = checkTaskExecution("task1")
+
+                taskLog.getExecutedTasks() shouldContainExactly setOf("task1")
+            }
+
+            "execute multiple tasks" {
+                val taskLog = checkTaskExecution("task1", "task2")
+
+                taskLog.getExecutedTasks() shouldContainExactlyInAnyOrder setOf("task1", "task2")
+            }
+
+            "handle errors during task execution" {
+                val taskLog = checkTaskExecution("task1", ERROR_TASK, "task2")
+
+                taskLog.getExecutedTasks() shouldContainExactlyInAnyOrder setOf("task1", ERROR_TASK, "task2")
+            }
+
+            "handle a non-existing task" {
+                val taskLog = checkTaskExecution("task1", "nonExistingTask", "task2")
+
+                taskLog.getExecutedTasks() shouldContainExactlyInAnyOrder setOf("task1", "task2")
+            }
+        }
+
+        "main" should {
+            "call runTasks with a database module" {
+                checkMain { koin ->
+                    verifyDatabaseModuleIncluded()
```

Comment:
Should this also verify that the `configModule` is included?

Reply:
This is implicitly tested by 'verifyDatabaseModuleIncluded()`; the function checks whether a `ConfigManager` is available.